src/client/components/Header.tsx

```diff
@@ -39,9 +66,43 @@ export const Header: React.FunctionComponent = (props) => {
             ? null :
               isLoggedIn
                 ? [
-                    <IconButton color="inherit" key={0}><SettingsIcon color="inherit"/></IconButton>,
+                    <IconButton color="inherit" key={0} onClick={() => setOpen(true)}><SettingsIcon color="inherit" /></IconButton>,
                     <Fragment key={1}>{data.me.name}</Fragment>,
-                    <Button onClick={logOutUser} key={2}>Log Out</Button>
+                    <Button onClick={logOutUser} key={2}>Log Out</Button>,
+                    <Modal open={open} onClose={()=> setOpen(false)} key={3} style={{
+                      display: 'flex',
+                      alignItems: 'center',
+                      justifyContent: 'center',
+                      
+                    }}>
+                      <Paper style={{
+                        minWidth: "500px",
+                        maxWidth: "800px",
+                        margin: "auto",
+                        padding: "10px"
+                        }}>
+                        <Grid container direction="column" alignContent="center" alignItems="center" spacing={1}>
+                          <Grid xs={12} item>
+                            Name: {data.me.name}
+                          </Grid>
+                          <Grid xs={12} item>
+                            Email: {data.me.email}
+                          </Grid>
+                            Change Password:
+                            <TextField id="old-password" label="Enter Old Password" autoFocus value={oldPassword} onChange={onTextFieldUpdate(setOldPassword)} ></TextField>
+                            <TextField id="new-password" label="Enter New Password" onChange={onTextFieldUpdate(setNewPassword)} helperText={!properPassword() ? 'Password must be between 9 and 255 characters' : ''} error={!properPassword()} type="password"></TextField>
+                            <TextField id="confirm-password" label="Confirm Password" onChange={onTextFieldUpdate(setConfirmPassword)} helperText={!areSame(newPassword, confirmPassword) ? 'Passwords do not match' : ''} error={!areSame(newPassword, confirmPassword)} type="password"></TextField>
+                            <Button onClick={() => handleConfirmPassword(oldPassword, newPassword, confirmPassword)}>
+                              Change Password
+                            </Button>
+                          <Grid xs={12} item>
+                            <Button onClick={() => setOpen(false)}>
+                              Close
+                            </Button>
+                          </Grid>
+                        </Grid>
+                      </Paper>
+                    </Modal>
```

Comment:
Move this all to the return statement of a new functional component called UserSettingsModal in `src/client/components/UserSettingsModal.tsx`. Also move across all of the corresponding variables that go along with this component except for the `const [open, setOpen] = useState(false);` since this should be two props that are passed into the UserSettingsModal component.